core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -734,12 +737,14 @@ abstract class RDD[T: ClassTag](
       printPipeContext: (String => Unit) => Unit = null,
       printRDDElement: (T, String => Unit) => Unit = null,
       separateWorkingDir: Boolean = false,
-      bufferSize: Int = 8192): RDD[String] = withScope {
+      bufferSize: Int = 8192,
+      encoding: Charset = StandardCharsets.UTF_8): RDD[String] = withScope {
```

Comment:
I suggest using `Codec.defaultCharsetCodec` as the default value, so that existing applications won't be broken.